src/Microsoft.OData.Core/UriParser/ExpressionLexer.cs

```diff
@@ -447,6 +453,24 @@ internal string AdvanceThroughBalancedParentheticalExpression()
             return expressionText;
         }
 
+        internal void TakeTextPositionSnapshot()

```

Comment:
Much better, thanks!

---
In reply to: [101113153](https://github.com/OData/odata.net/pull/757#discussion_r101113153) [](ancestors = 101113153)